modules/ui/commit.js

```diff
@@ -507,12 +507,11 @@ export function uiCommit(context) {
             detectedHashtags = detectedHashtags.concat(hashtagHashtags());
         }
 
-        var allLowerCase = new Set();
+        var checkHashTag = new Set();
         return detectedHashtags.filter(function(hashtag) {
-            // Compare tags as lowercase strings, but keep original case tags
-            var lowerCase = hashtag.toLowerCase();
-            if (!allLowerCase.has(lowerCase)) {
-                allLowerCase.add(lowerCase);
+            // keep only unique hashtags
+            if (!checkHashTag.has(hashtag)) {
+                checkHashTag.add(hashtag);
```

Comment:
Are you sure this change fixes #10038? The function seems to discard the ~~`allLowercase`~~ `checkHashTag` set anyways. All it uses the set for is to determine whether we’ve previously seen the hashtag case-insensitively as we filter `detectedHashtags`, to prevent the resulting hashtags from including more than one case variant, such as `hashtag=HappyMapping;happyMapping;happymapping`. https://github.com/openstreetmap/iD/issues/10038#issuecomment-1868331583 suggests a different fix.